pkg/kubelet/util/pluginwatcher/plugin_watcher_test.go

```diff
@@ -238,11 +238,22 @@ func TestPlugiRegistrationFailureWithUnsupportedVersionAtKubeletStart(t *testing
 	hdlr := NewExampleHandler(supportedVersions, false /* permitDeprecatedDir */)
 	hdlr.AddPluginName(pluginName)
 
+	c := make(chan struct{})
+	go func() {
+		defer close(c)
+		require.True(t, waitForEvent(t, exampleEventValidate, hdlr.EventChan(p.pluginName)))
+		require.False(t, waitForPluginRegistrationStatus(t, p.registrationStatus))
+	}()
```

Comment:
alternatively, you can just do this serially at the end of the test and rely on the whole test timing out for the failure condition.

Reply:
Since we're processing plugins serially now, `newWatcherWithHandler` below would block forever trying to write to the event/registration channels. This goroutine prevents that by receiving info from these channels beforehand.